contracts/BlendSwap.sol

```diff
@@ -38,80 +39,91 @@ contract BlendSwap {
     }
 
     function lock(
-        bytes32 lockId,
         address to,
         uint256 amount,
         uint releaseTime,
-        bytes32 secretHash
+        bytes32 secretHash,
+        bool confirmed,
+        uint256 fee
     )
         public
     {
         require(
-            status[lockId] == Status.NOT_INITIALIZED,
-            "Lock with this id already exists"
+            status[secretHash] == Status.NOT_INITIALIZED,
+            "Lock with this secretHash already exists"
         );
 
-        swaps[lockId] = Swap({
+        swaps[secretHash] = Swap({
             from: msg.sender,
             to: to,
             amount: amount,
-            releaseTime: releaseTime
+            releaseTime: releaseTime,
+            secretHash: secretHash,
+            fee: fee
         });
 
-
-        if (secretHash == 0x00) {
-            status[lockId] = Status.INITIALIZED;
+        if (confirmed) {
+            status[secretHash] = Status.CONFIRMED;
         } else {
-            status[lockId] = Status.HASH_REVEALED;
-            hashlocks[lockId] = secretHash;
+            status[secretHash] = Status.INITIALIZED;
         }
 
-        blend.transferFrom(msg.sender, address(this), amount);
+        blend.transferFrom(msg.sender, address(this), amount + fee);
```

Comment:
I think it is a question for @kkirka

Reply:
Idk what `sLockSaver` is on Tezos, here all the locked tokens are "owned" by the BlendSwap contract, so that BlendSwap can ERC20 `transfer` them

Reply:
This is because _Tezos_ have view method probably, and thus can execute arbitrary code on contract behalf, but I don't remember already what was the problem exactly